pkg/webhook/controlplane/ensurer.go

```diff
@@ -337,45 +376,94 @@ func ensureKubeAPIServerCommandLineArgs(c *corev1.Container, makeAuditForwarder
 
 }
 
+func (e *ensurer) getCustomAuditPolicy(ctx context.Context, cluster *extensions.Cluster) (bool, error) {
+	if extensionscontroller.IsHibernated(cluster) { // Cluster is hibernated - no Apiserver to ask
+		return false, nil
+	}
+
+	shootConfig, _, err := util.NewClientForShoot(ctx, e.client, cluster.ObjectMeta.Name, client.Options{})
+	if err != nil {
+		return false, err
+	}
+
+	cs, err := kubernetes.NewForConfig(shootConfig)
+	if err != nil {
+		return false, err
+	}
+
+	customAuditPolicyCm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "custom-audit-policy",
+			Namespace: cluster.ObjectMeta.Name,
+		},
+	}
+
+	currentCustomAuditPolicy := &corev1.ConfigMap{}
+	err = e.client.Get(ctx, types.NamespacedName{Namespace: cluster.ObjectMeta.Name, Name: "custom-audit-policy"}, currentCustomAuditPolicy)
+	if err != nil {
+		logger.Info("AUDITDEBUG no custom auditpolicy in shoot namespace")
+	}
+
+	customAuditPolicyShootCm, _ := cs.CoreV1().ConfigMaps("kube-system").Get(ctx, "custom-audit-policy", metav1.GetOptions{})
+	if customAuditPolicyShootCm == nil || customAuditPolicyShootCm.Name != "custom-audit-policy" {
+		if currentCustomAuditPolicy.Name == "custom-audit-policy" {
+			logger.Info("AUDITDEBUG no custom auditpolicy configmap in shoot, deleting custom auditpolicy configmap in shoot namespace", "configmap", customAuditPolicyCm)
+			err := e.client.Delete(ctx, customAuditPolicyCm)
+			if err != nil {
+				return false, err
+			}
+		}
+		return false, nil
+	}
+
+	logger.Info("AUDITDEBUG found custom auditpolicy configmap in shoot", "shoot-configmap", customAuditPolicyShootCm)
```

Comment:
log prefix is weird, simple remove the "AUDITDEBUG"

Reply:
The AUDITDEBUG messages were just for making sure the code does what I think it does - I'll pare them down to a less verbose logging when it's ready for review.